### PR TITLE
[Proof of Concept] Vitejs implementation

### PR DESCRIPTION
**Pull request by willwill96**
More information on vitejs plugins API here:
https://vitejs.dev/guide/api-plugin.html#plugin-api

### CODE REVIEW

**Comment by garronej** [comment]
🚀

File: vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+const htmlPlugin = ()=>{
+    return {
+        name:'html-plugin',
+        transformIndexHtml(html){
+            return html.replace('/assets','${url.resourcesPath}/build/static')
+        }
+    }
+}
+function replaceImportsInCssCode(params: { cssCode: string }): {
```
Comment (willwill96) [inline vite.config.ts:15]: copied directly from keycloakify

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    Object.keys(cssGlobalsToDefine).forEach(
+        cssVariableName =>
+            //NOTE: split/join pattern ~ replace all
+            (fixedCssCode = fixedCssCode.split(cssGlobalsToDefine[cssVariableName]).join(`var(--${cssVariableName})`))
+    );
+
+    return { fixedCssCode, cssGlobalsToDefine };
+}
+
+function generateCssCodeToDefineGlobals(params: { cssGlobalsToDefine: Record<string, string>; }): {
```
Comment (willwill96) [inline vite.config.ts:38]: copied directly from keycloakify with the exception of removing buildOptions

File: vite.config.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                    "",
+                    "<style>",
+                    generateCssCodeToDefineGlobals({
+                        cssGlobalsToDefine,
+                    }).cssCodeToPrependInHead,
+                    "</style>",
+                    ""
+                ].join("\n")
+            );
+            return $.html();
```
Comment (willwill96) [inline vite.config.ts:94]: This logic is responsible for replacing absolute css urls to css vars that are injected into the head (`transformIndexHtml` logic copied from keycloakify)

File: vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        outDir:'build',
+        target:'es2015',
+        assetsDir: 'static'
+    },
+    experimental: {
+        renderBuiltUrl(filename, {hostType}) {
+            if ((hostType) === 'js')
+                return { runtime: `window.${ftlValuesGlobalName}.url.resourcesPath + "/build/" + ${JSON.stringify(filename)}`}
+            return { relative: true }
+        }
```
Comment (willwill96) [inline vite.config.ts:117]: More docs here: https://vitejs.dev/guide/build.html#advanced-base-options

But this effectively replicates the functionality found here in keycloakify: https://github.com/InseeFrLab/keycloakify/blob/main/src/bin/keycloakify/replacers/replaceImportsFromStaticInJsCode.ts#L28

**Pull request closed without merging**